_getline.c

```diff
@@ -0,0 +1,106 @@
+#include "shell.h"
+
+/**
+ * _getline - read a line from a file descriptor and allocate memory for it.
+ * @fd: the file descriptor from which to read the line.
+ * @line: a pointer to a char pointer where the resulting line will be stored.
+ *
+ * This function reads a line from the specified file descriptor and allocates
+ * memory to store the line. The memory for the line should be freed by the
+ * caller when it is no longer needed.
+ *
+ * Return: Upon success, the function returns the number of characters read,
+ * excluding the null-terminator. On end of file or error, it returns -1.
+ */
+
+int _getline(char **line, const int fd)
+{
+	static char *rd = NULL;
+	char buffer[SIZE];
+	int pos;
+
+	pos = _strchr(rd, '\n');
+
+	if (pos >= 0)
+	{
+		process_line(&*line, &rd, pos);
+		return (pos);
+	}
+
+	while ((pos = _read(fd, &rd, buffer)) >= 0)
```

Comment:
can you use another variable than pos, each variable should have one function or job to do, this way everything will be easy to read and understand, try to declare an `int r` for the read return value.